pkg/compute/kube/client.go

```diff
@@ -549,3 +553,84 @@ func (c *client) ListNamespaces() ([]string, error) {
 	}
 	return names, nil
 }
+
+// ListTrafficSplits implements mesh.MeshSpec by returning the list of traffic splits.
+func (c *client) ListTrafficSplitsByOptions(options ...smi.TrafficSplitListOption) []*smiSplit.TrafficSplit {
+	var trafficSplits []*smiSplit.TrafficSplit
+
+	for _, trafficSplit := range c.kubeController.ListTrafficSplits() {
```

Comment:
a good rule of thumb: if this method can be placed in the mesh catalog with no changes it should.

That is, we can leverage the passthrough interface, and have this method simply on the catalog itself.

Then we can even unexport it if it's not used elsewhere.

Reply:
The reason for this, is that the higher level functions, that don't need knowledge about k8s, can be used out of the compute client, and re-used with another client. This makes it more simple to plug in a new interface

Reply:
This would further allow us to unexport all the types defined in SMI, bring those types into the catalog package, and remove the SMI package altogether